object/src/mem.rs

```diff
@@ -758,6 +785,45 @@ where
         self.entries.insert(elt.tag(), elt)
     }
 
+    /// Insert a private element into the dataset, replacing (and returning) any
+    /// previous element of the same attribute.
+    ///
+    /// This function will find the next available private element block in the given
+    /// group. If the creator already exists, the element will be added to the block
+    /// already reserved for that creator. If it does not exist, then a new block
+    /// will be reserved for the creator in the specified group.
+    ///
+    /// For more info, see the [DICOM standard section on private elements](https://dicom.nema.org/medical/dicom/2024a/output/chtml/part05/sect_7.8.html)
+    ///
+    /// ## Example
+    /// ```
+    /// # use dicom_core::{VR, PrimitiveValue, Tag, DataElement, header::Header};
+    /// # use dicom_object::InMemDicomObject;
+    /// let mut ds = InMemDicomObject::new_empty();
+    /// ds.put_private_element(
+    ///     0x0009,
+    ///     "CREATOR 1",
+    ///     0x02,
+    ///     VR::DS,
+    ///     PrimitiveValue::Str("1.0".to_string()),
+    /// )
+    /// .unwrap();
+    /// assert_eq!(
+    ///     ds.private_element(0x0009, "CREATOR 1", 0x02)
+    ///         .unwrap()
```

Comment:
Examples should prefer `?` for raising errors.

Reply:
So I found [this](https://doc.rust-lang.org/rustdoc/write-documentation/documentation-tests.html#using--in-doc-tests), but I'm not sure how to get it to work without using `map_err` since I have two different error types (`PrivateElementError` and `ConvertValueError`). Any suggestions?

Reply:
Ended up getting it to work with `# Ok::<(), Box<dyn Error>>(())`.

Another question for you.  I wanted to format the code in the examples in doctests.  I found I can run `cargo fmt -- --config format_code_in_doc_comments=true object/src/mem.rs` but that this reformatted much more than I touched.  Do you have a normal way you do this?

Reply:
In those cases I just use `Box<dyn std::error::Error>` as the almost-universal common ground.